utils/roles.py

```diff
@@ -0,0 +1,71 @@
+import discord
+from bot_globals import MILESTONE_ROLES, STREAK_ROLES, logger, client
+
+async def give_user_streak_role(user: discord.User, guild_id: int, streak: int) -> None:
+    logger.info(f"file: utils/stats.py ~ give_user_streak_role ~ run - {user.id} - {streak}")
+    guild = client.get_guild(guild_id)
+
+    if not guild:
+        return
+    
+    discord_user = guild.get_member(user.id)
+
+    if not discord_user:
+        return
+    
+    role_to_assign = None
+    
+    for role_streak, (role_name, role_color) in STREAK_ROLES.items():
```

Comment:
You use tabs not spaces here, as every file in this project uses spaces and we don't want issues to arise which actually could. I'd suggest setting the default to spaces. In vs code you can also make it automatically convert a tab to 4 spaces (which may be by default tho). You will see a lot of changes to this file when I push my commit, which will be due to this issue.